plasmapy/physics/parameters.py

```diff
@@ -0,0 +1,1486 @@
+s) and angular velocity (rad / s).  A clear way to do this
+conversion is to set up an equivalency between cycles/s and Hz:
+
+>>> from astropy import units
+>>> f_ce = omega_ce.to(units.Hz, equivalencies=[(units.cy/units.s, units.Hz)])
+
+However, dimensionless_angles() does work when dividing a velocity
+by an angular frequency to get a length scale:
+
+>>> d_i = (c/omega_pi).to(units.m, equivalencies=units.dimensionless_angles())
+
+"""
+
+
+def Alfven_speed(B, density, ion="p"):
+    r"""Returns the Alfven speed.
+
+    Parameters
+    ----------
+    B : Quantity
+        The magnetic field magnitude in units convertible to tesla.
+
+    density: Quantity
+        Either the ion number density in units convertible to 1 / m**3,
+        or the mass density in units convertible to kg / m**3.
+
+    ion : string, optional
+        Representation of the ion species.  If not given, then the ions
+        are assumed to be protons.
+
+    Returns
+    -------
+    V_A : Quantity with units of velocity
+        The Alfven velocity of the plasma in units of meters per second.
+
+    Raises
+    ------
+    TypeError
+        The magnetic field and density arguments are not Quantities.
+
+    UnitConversionError
+        If the magnetic field or density is not in appropriate units.
+
+    UserWarning
+        If the Alfven velocity exceeds 10% of the speed of light, or
+        if units are not provided and SI units are assumed.
+
+    ValueError
+        If the density is negative, or the ion mass or charge state
+        cannot be found.
+
+    Notes
+    -----
+    The Alfven velocity :math:`V_A` is the typical propagation speed
+    of magnetic disturbances in a plasma, and is given by:
+
+    .. math::
+    V_A = \frac{B}{\sqrt{\mu_0\rho}}
+
+    where the mass density is :math:`\rho = n_i m_i + n_e m_e`.
+
+    This expression does not account for relativistic effects, and
+    loses validity when the resulting speed is a significant fraction
+    of the speed of light.
+
+    This function switches B and density when B has units of number
+    density or mass density and density has units of magnetic field
+    strength.
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> from plasmapy.constants import m_p, m_e
+    >>> B = 0.014*u.T
+    >>> n = 5e19*u.m**-3
+    >>> rho = n*(m_p+m_e)
+    >>> ion = 'p'
+    >>> Alfven_speed(B, n, ion)
+    <Quantity 43173.871857213584 m / s>
+    >>> Alfven_speed(B, rho, ion)
+    <Quantity 43173.871857213584 m / s>
+    >>> Alfven_speed(rho, B, ion)
+    <Quantity 43173.871857213584 m / s>
+    >>> Alfven_speed(B, rho).to(u.cm/u.us)
+    <Quantity 4.317387185721358 cm / us>
+
+    """
+
+    if isinstance(B, units.Quantity) and isinstance(density, units.Quantity):
+        if B.si.unit in [units.m**-3, units.kg/units.m**3] and \
+                density.si.unit in [units.T]:
+            B, density = density, B
+
+    _check_quantity(B, 'B', 'Alfven_speed', units.T)
```

Comment:
We may have been over this, but why are you using a `_check_quantity` function rather than the astropy `@quantity` decorator?